packages/shared/src/UI/components/SettingsBoard/GasForm.tsx

```diff
@@ -0,0 +1,246 @@
+import { useEffect, useMemo } from 'react'
+import { useUpdateEffect } from 'react-use'
+import { Controller, FormProvider, useForm } from 'react-hook-form'
+import BigNumber from 'bignumber.js'
+import { makeStyles, MaskAlert, MaskColorVar, MaskTextField } from '@masknet/theme'
+import { Grid, Typography } from '@mui/material'
+import { WarningIcon } from '@masknet/icons'
+import { useSharedI18N } from '@masknet/shared'
+import {
+    ChainId,
+    formatGweiToEther,
+    formatGweiToWei,
+    formatWeiToGwei,
+    GasOption,
+    Transaction,
+} from '@masknet/web3-shared-evm'
+import { GasOptionType, NetworkPluginID } from '@masknet/web3-shared-base'
+import { useWeb3State } from '@masknet/plugin-infra/web3'
+import type { z as zod } from 'zod'
+import { zodResolver } from '@hookform/resolvers/zod'
+import { useGasSchema } from './hooks'
+
+const useStyles = makeStyles()((theme) => {
+    return {
+        unit: {
+            color: MaskColorVar.textLight,
+        },
+        textfield: {
+            '& input[type=number]': {
+                '-moz-appearance': 'textfield',
+            },
+            '& input[type=number]::-webkit-outer-spin-button': {
+                '-webkit-appearance': 'none',
+            },
+            '& input[type=number]::-webkit-inner-spin-button': {
+                '-webkit-appearance': 'none',
+            },
+        },
+        caption: {
+            color: MaskColorVar.textSecondary,
+            fontWeight: 700,
+            margin: theme.spacing(1, 0, 1.5),
+        },
+    }
+})
+
+export interface GasFormProps {
+    chainId: ChainId
+    transaction: Transaction
+    gasOptions: Record<GasOptionType, GasOption>
+    onChange?: (transactionOptions?: Partial<Transaction>) => void
+}
+
+export function GasForm(props: GasFormProps) {
+    const { chainId, transaction, gasOptions, onChange } = props
+    const t = useSharedI18N()
+    const { classes } = useStyles()
+    const { Others } = useWeb3State(NetworkPluginID.PLUGIN_EVM)
+
+    const isEIP1559 = Others?.chainResolver.isSupport(chainId, 'EIP1559')
+
+    const schema = useGasSchema(chainId, transaction, gasOptions)
+
+    const methods = useForm<zod.infer<typeof schema>>({
+        shouldUnregister: false,
+        mode: 'onChange',
+        resolver: zodResolver(schema),
+        defaultValues: {
+            gasLimit: (transaction.gas as string | undefined) ?? '210000',
```

Comment:
The default value should be 21000. Is this the default value for non-evm networks as well? If not I think we need to define a constant in a different network.

Reply:
It's an EVM-only form.